cli/src/trades.rs

```diff
@@ -23,8 +23,12 @@ pub fn all(client: &Client, matches: &ArgMatches) -> Result<()> {
         if let (Some(base_str), Some(counter_str)) =
             (matches.value_of("base"), matches.value_of("counter"))
         {
-            let base = asset_identifier::from_str(base_str)?;
-            let counter = asset_identifier::from_str(counter_str)?;
+            let base: AssetIdentifier = base_str
+                .parse::<AssetIdentifier>()
+                .map_err(|_| String::from("Base asset must be properly formatted asset"))?;
+            let counter: AssetIdentifier = counter_str
```

Comment:
In some of these places you have the type specified more than once. You should only need to add the type either next to the variable `let counter: AssetIdentifier` or as part of the parse function `.parse::<AssetIdentifier>`. You can safely remove one.

Reply:
Going with the second to keep everything consistent